From f1e7b31b5e0cee14894baa644aa1470fe90a607d Mon Sep 17 00:00:00 2001
From: Christine Gharzuzi <cgharzuzi@habana.ai>
Date: Mon, 17 Feb 2020 14:47:54 +0200
Subject: [PATCH] [SW-0] goya2: use invert dword in spi otp

Invert is also needed in spi otp same as in
spi flash.

Change-Id: I40bdade5df2e505a16d8017a7615fe81a4ed42a2
Signed-off-by: Christine Gharzuzi <cgharzuzi@habana.ai>
---

diff --git a/common/spi_flash.c b/common/spi_flash.c
index c826440..9058aec 100644
--- a/common/spi_flash.c
+++ b/common/spi_flash.c
@@ -113,6 +113,8 @@
 {
 	*key_revoke = spi_flash_readl(FLASH_EFUSE_KEYS_REVOKE_OFFSET) &
 			SECURITY_KEY_REVOKE_MAP;
+
+	spi_flash_invert_dword(key_revoke);
 }
 
 void spi_flash_otp_get_hash(uint32_t *key_hash)
@@ -131,8 +133,10 @@
 	uint32_t flash_efuse_offset = FLASH_OTP_FW_VERSION_OFFSET;
 
 	/* read min security version bit map from flash */
-	for (i = 0; i < SECURITY_DWORD_PER_FW_SEC_VERSION; i++)
+	for (i = 0; i < SECURITY_DWORD_PER_FW_SEC_VERSION; i++) {
 		fw_version[i] = spi_flash_readl(flash_efuse_offset + i);
+		spi_flash_invert_dword(&fw_version[i]);
+	}
 
 	fw_version_sec = security_find_min_fw_version(fw_version);
 	return fw_version_sec;
